File: DM/1_HW/7.py
from itertools import product


n , m = [int(x) for x in input().split()]
graph = [set() for x in range(n)]

for i in range(m):
    start, end = [int(x) for x in input().split()]
    graph[start].add(end)
    graph[end].add(start)



for pr in product(range(n), repeat=n):
    if len(set(pr)) == len(pr):
        diff = set()
        flag = False
        visited = [False] * n
        for i in range(n):
            visited[i] = True
            
            for nv in graph[i]:
                if (not visited[nv]):
                    diff.add(abs(pr[i] - pr[nv]))
        if (len(diff) == m):
            print(pr)

    

File: DM/1_HW/7.cpp
#include <iostream>
#include <set>

void task_7() {
    int n, m;
    std::cin >> n >> m;

    std::set<int>* graph = new std::set<int>[n];

    for (int i = 0; i < m; ++i) {
        int start, end;
        std::cin >> start >> end;

        graph[start].insert(end);
        graph[end].insert(start);
    }

    
}   